### ISSUE TITLE
Not compatible with Tab Mix Plus?

### ISSUE THREAD

**Issue by ghost** [vovcacik/bug-566510-firefox-addon#1]
Firefox 26b1 on win XP. Not work when Tab Mix Plus is enabled - work without it.

**Comment by vovcacik** [comment]
Hi, it is possible but I am not very happy about it. 

#### Dirty way

TMP is also using XBL to extend tab functionality, but multiple inheritance is not supported in XBL 1.0. That's why only one of the addons is loaded (actually it is always TMP as it uses !important CSS flag on the -moz-binding property).

I could rewrite this addon to extend TMP bindings instead of the default one, which I did in issues/1 branch. And it seems to work fine. You can try it if it works for you:

https://github.com/vovcacik/bug-566510-firefox-addon/blob/issues/1/bug_566510_allow_multiselect_operations_on_tabs-0.12.3-fx.xpi

**Don't install it in your main Firefox profile, as something may go wrong and you end up with corrupted data.**
This way I could use something like _setProperty(propertyName, value, priority)_ to force my binding, but thats quite nasty; lot of code repetition. Not likely to merge it in master.

#### Clean way

MDN documents [a technique](https://developer.mozilla.org/en-US/docs/XBL/XBL_1.0_Reference/Binding_Attachment_and_Detachment#Attachment_using_element.style_property) that would nicely solve this. But it does not work and I was told the MDN page is completely wrong.

**Comment by Keith94** [comment]
I talked to the TMP developer about this before, he said he can add an option like "control with another extension" for its mouse clicking commands, which should allow these to be compatible.

**Comment by vovcacik** [comment]
@Keith94, actually that's not a problem. I've briefly looked into TMP source and it seems there are no conflicts with this addon - except ctrl+click and shift+click. But you would just unbind those combinations in TMP settings (I think the default binding is marking tab for merging and locking tab).

**Comment by Keith94** [comment]
I know, but that's why I asked the TMP dev to implement a new option that would allow other extensions to use ctrl/shift click operations without any conflicts, so then the build you posted above wouldn't be necessary. At least I assumed that idea would work anyway.

**Comment by vovcacik** [comment]
The shift and ctrl conflict is not the reason for the extra build. The
problem is XBL bindings are attached to XUL elements with -moz-binding CSS
property:

``` css
.tabbrowser-tab {
    -moz-binding:
url("chrome://bug566510/content/tabbrowserOverlay.xml#bug566510_tabbrowser-tabOverlay");
}
```

``` css
.tabbrowser-tab {
  -moz-binding:
url("chrome://tabmixplus/content/tab/tabbrowser_4.xml#tabmix-tabbrowser-tab")
!important;
}
```

This is of course a conflict which will be resolved only in favor of one of
the bindings. In our case TMP got the highest priority so bug566510
bindings won't be loaded at all.

**Comment by vovcacik** [comment]
After some thoughts on XBL inheritance I was able to come up with something that works - XBL proxy. It neatly solves the conflict and requires minimal changes.

Keep an eye on [pull request](https://bitbucket.org/onemen/tabmixplus/pull-request/1/compatibility-with-bug566510-addon-via-xbl/diff) and the [issues/1-proxy](https://github.com/vovcacik/bug-566510-firefox-addon/tree/issues/1-proxy) branch.

**Comment by vovcacik** [comment]
Fixed since TabMixPlus [v0.4.1.3pre.131110a1](https://bitbucket.org/onemen/tabmixplus/downloads#download-262347) and Bug 566510 [v0.12.4](https://addons.mozilla.org/en-us/firefox/addon/bug-566510/versions/0.12.4)
